src/util/simplify_expr_int.cpp

```diff
@@ -1178,6 +1200,9 @@ bool simplify_exprt::simplify_inequality(exprt &expr)
   exprt tmp0=expr.op0();
   exprt tmp1=expr.op1();
 
+  if(!simplify_constant_equality(expr))
+    return false;
+
```

Comment:
Isn't this placement of the call somewhat confusing as we (also) handle the case `op0_is_const && op1_is_const` below?

Reply:
Inlined it further down now.